docs/lib/storage/fragments/ios/configureaccess/50_customization.md

```diff
@@ -0,0 +1,136 @@
+            }
+            return
+        }
+
+        getIdentityId { result in
+            switch result {
+            case .failure(let error):
+                completion(.failure(error))
+            case .success(let identityId):
+                if accessLevel == .protected {
+                    completion(.success("myProtectedPrefix/" + identityId + "/"))
+                } else if accessLevel == .private {
+                    completion(.success("myPrivatePrefix/" + identityId + "/"))
+                }
+
+            }
+        }
+    }
+
+    public func getIdentityId(completion: @escaping (Result<String, StorageError>) -> Void) {
+        Amplify.Auth.fetchAuthSession { result in
+            do {
+                let session = try result.get()
+                if let identityProvider = session as? AuthCognitoIdentityProvider {
+                    let identityId = try identityProvider.getIdentityId().get()
+                    completion(.success(identityId))
+                }
+            } catch {
+                completion(.failure(StorageError.authError("Failed to retrieve auth session", "", error)))
+            }
+        }
+    }
+}
+```
+
+Then configure the storage plugin with the resolver.
+
+```swift
+let storagePlugin = AWSS3StoragePlugin(configuration: .prefixResolver(MyDeveloperDefinedPrefixResolver()))
+Amplify.add(storagePlugin)
+```
+
+For example, if you want to enable read, write and delete operation for all the objects under path *myPublicPrefix/*, declare it in your IAM policy:
+
+```xml
+"Statement": [
+    {
+        "Effect": "Allow",
+        "Action": [
+            "s3:GetObject",
+            "s3:PutObject",
+            "s3:DeleteObject"
+        ],
+        "Resource": [
+            "arn:aws:s3:::your-s3-bucket/myPublicPrefix/*",
+        ]
+    }
+]
+```
+
+If you want to have custom *private* path prefix like *myPrivatePrefix/*, you need to add it into your IAM policy:
+```xml
+"Statement": [
+    {
+        "Effect": "Allow",
+        "Action": [
+            "s3:GetObject",
+            "s3:PutObject",
+            "s3:DeleteObject"
+        ],
+        "Resource": [
+            "arn:aws:s3:::your-s3-bucket/myPrivatePrefix/${cognito-identity.amazonaws.com:sub}/*"
+        ]
+    }
+]
+```
+This ensures only the authenticated users has the access to the objects under the path. 
+
+#### Examples
+
+1. See [StorageAccessLevelAwarePrefixResolver](https://github.com/aws-amplify/amplify-ios/blob/main/AmplifyPlugins/Storage/AWSS3StoragePlugin/Configuration/AWSS3PluginPrefixResolver.swift#L26) and [StorageRequestUtils.getAccessLevelPrefix](https://github.com/aws-amplify/amplify-ios/blob/main/AmplifyPlugins/Storage/AWSS3StoragePlugin/Support/Utils/StorageRequestUtils+Getter.swift#L24) for an example of the prefix resolver used by the plugin. You do not have to add this yourself and is used by default if no prefix resolver is added to the configuration.
```

Comment:
Giving code reference like this would break in future when we update the code. Maybe use permalinks?

Reply:
what do you mean by using a permalink? The feedback I got was to explain the current logic since it would be useful for developers looking into custom resolvers. I think i can address this by explaining it without linking to code snippet

Reply:
as discussed, will describe the example instead of relying on code snippet here